backend/src/routes/api/userPatch.routes.js

```diff
@@ -0,0 +1,44 @@
+import { StatusCodes } from 'http-status-codes';
+import User, { userValidator } from '../../models/User.js';
+
+const userPatchEndpoint = (router) => {
+  router.patch('/users/:id', getUser, userValidator, async (req, res) => {
+    res.user_patch = {};
+    if (req.body.username != null) {
+      res.user_patch.username = req.body.username;
+    }
+    if (req.body.email != null) {
+      res.user_patch.email = req.body.email;
+    }
+    if (req.body.password != null) {
+      res.user_patch.password = req.body.password;
+    }
```

Comment:
I'm wondering if we need this part here - req.body should be validated by validation middleware by now. We should avoid such an approach, because 
a) we need to remember about extending it every time model is extended
b) with bigger models this part would be sooo long :D

Reply:
Yes, I am using validation middleware. It was not for this reason. Previously, I had this function written differently and it was actually needed, but after the change I made later, it really doesn't seem to be needed.